draft.markdown

```diff
@@ -0,0 +1,155 @@
+# Statistical Aggregations
+
+* Target Audience: Developers
+* Article Type: Development Frameworks
+
+## External References
+
+* https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-aggregations-metrics.html
+* https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-aggregations-metrics-cardinality-aggregation.html
+* https://lucene.apache.org/solr/guide/7_5/the-stats-component.html
+
+## Search Engine Connector Support
+* Elasticsearch 6: Yes
+* Solr 7: Yes
+
+## New/Related APIs
+
+Listed those which are relevant for developers.
+ 
+API (FQCN) | Provided by Artifact |
+---------: | :------------------: |
+`com.liferay.portal.search.searcher.SearchRequestBuilder#statsRequests(StatsRequest... statsRequests)` | com.liferay.portal.search.api
+`com.liferay.portal.search.searcher.SearchResponse#getStatsResponseMap()` | com.liferay.portal.search.api
+**`com.liferay.portal.search.stats.StatsRequest`** |	com.liferay.portal.search.api
+`com.liferay.portal.search.stats.StatsRequestBuilder` |	com.liferay.portal.search.api
+`com.liferay.portal.search.stats.StatsRequestBuilderFactory` |	com.liferay.portal.search.api
+**`com.liferay.portal.search.stats.StatsResponse`** |	com.liferay.portal.search.api
+`com.liferay.portal.kernel.search.Stats` | portal-kernel
+
+## Deprecated APIs
+* SearchSearchRequest#getStats()
+* SearchSearchRequest#setStats(Map<String, Stats> stats)
+
+----
+
+## Draft
+
+Support for [GroupBy](https://github.com/liferay/liferay-portal/blob/7.2.x/portal-kernel/src/com/liferay/portal/kernel/search/GroupBy.java) and [Stats](https://github.com/liferay/liferay-portal/blob/7.2.x/portal-kernel/src/com/liferay/portal/kernel/search/Stats.java) aggregations were introduced in 7.0.
+
+To extend Liferay's search aggregation capabilities, in @product-ver@, we added support for Cardinality Aggregations which provides an approximate (aka. statistical) count of distinct values. Please refer to the "External References" above for more details.
+
+As part of the modularization efforts, [StatsRequest](https://github.com/liferay/liferay-portal/blob/7.2.x/modules/apps/portal-search/portal-search-api/src/main/java/com/liferay/portal/search/stats/StatsRequest.java) and [StatsResponse](https://github.com/liferay/liferay-portal/blob/7.2.x/modules/apps/portal-search/portal-search-api/src/main/java/com/liferay/portal/search/stats/StatsResponse.java) were introduced in the `com.liferay.portal.search.api` module to avoid modifying `portal-kernel`. StatsRequest provides the same statistical features that the legacy `com.liferay.portal.kernel.search.Stats` does besides the new "cardinality" option.

```

Comment:
@lipusz, are we saying here that the StatsRequest and StatsResponse replaces GroupBy and Stats?

Reply:
Maybe a better question is, does anything replace GroupBy?

Reply:
GroupBy is a TermsAggregation: https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-aggregations-bucket-terms-aggregation.html

https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-search-elasticsearch6/portal-search-elasticsearch6-impl/src/main/java/com/liferay/portal/search/elasticsearch6/internal/groupby/DefaultGroupByTranslator.java#L64

So the answer is no, we are not replacing GroupBy, it's a different aggregation type.